examples/analogin/main.c

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2021 Arm Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ #include "bootstrap/mbed_trace.h"
+ #include "bootstrap/mbed_wait_api.h"
+ #include "hal/analogin_api.h"
+ #include "hal/serial_api.h"
+
+ #define VOLTAGE_MULTIPLIER 3.3
+
+ static serial_t stdio_uart;
+
+ void serial_print(const char *string)
+ {
+     while (*string != '\0') {
+         serial_putc(&stdio_uart, *string);
+         string++;
+     }
+     serial_putc(&stdio_uart, '\n');
+ }
+
+ int main()
```

Comment:
```suggestion
 int main(void)
```
The use of function declarators with empty parentheses (not prototype-format parameter type declarators) is an obsolescent feature. Functions that accept no arguments should explicitly declare a `void` parameter in their parameter list.

Reply:
`main()` is a bit special but no argument at all is bad (see [this](https://godbolt.org/z/P3W4E1vbq)).

Reply:
Oh yes I know it is special. You can also omit the return statement even if there is a non-void return type but for consistency sake I would just follow the same rule.

Reply:
That is always good fun and exercise (to me) digging back the c specs,
6.7.6.3 §14

> […]. The  empty  list  in  a  function  declarator  that  is  not  part  of  a definition of that function specifies that **no information about the number or types of the parameters is supplied.**